tests/datastore/test_filestore.py

```diff
@@ -45,3 +45,21 @@ def test_rm_file_not_found(self, prefix):
         not_exist_url = f"{prefix}/path/to/file/not_exist_file.txt"
         data_item = mlrun.run.get_dataitem(not_exist_url)
         data_item.delete()
+
+    @pytest.mark.parametrize("data", [b"test", bytearray(b"test")])
+    def test_put_types(self, data, prefix):
+        try:
+            with tempfile.NamedTemporaryFile(suffix=".txt", delete=False) as temp_file:
```

Comment:
Why not let it do the delete for you?
```suggestion
            with tempfile.NamedTemporaryFile(suffix=".txt") as temp_file:
```